src/plugins/janus_streaming.c

```diff
@@ -548,6 +548,26 @@ multistream-test: {
 {
 	"streaming" : "ok"
 }
+\endverbatim
+ *
+ * You can kick all viewers from a mountpoint using the \c kick_all request. Notice
+ * that this only removes all viewers, but does not prevent them from starting to watch
+ * the mountpoint again. The \c kick_all request has to be formatted as follows:
```

Comment:
I think this request is unneeded: we already have a `disable` request that can be used for this, and maybe just needs to be extended to close the PeerConnections when we're disabling a mountpoint. This gives you time to change the PIN and then re-enable the mountpoint.

Reply:
Currently, the `disable` request seems to be for a totally different use case. As far as I understand, it does not explicitly close the peer connections, but just disables the stream -- though I have to say I didn't test what really happens at that point to the viewers.

Anyway, with this use case, I really would not like to disable the endpoint at all.

Also, from the developer's viewpoint, having symmetric/similar requests in plugins is helpful IMO. And when thinking about the naming the meaning of the requests, having "kick all" as an option of a `disable` request sounds a bit artificial.

As a summary, I feel like `disable` and `kick_all` are two entirely different pieces of functionality and they complement each other rather than being similar.

Thoughts?

Reply:
The Streaming plugin is completely different from AudioBridge, VideoRoom and TextRoom: it doesn't have a concept of room at all, so I think "kick" makes little sense. Since the context is completely different, symmetric requests are less important IMO.

It makes much more sense to me to disable a mountpoint and have that feature optionally close PeerConnections (which is something a few asked for in the past, I can't remember why that didn't happen eventually): since what you're trying to do is kicking everyone out and preventing them access until you change something, "disable" is actually a much better fit.

Reply:
Ok, so could the `disable` request then contain options for:

1) closing all connections (not calling it kicking anymore), e.g. `close_connections` - defaults to `false` to preserve existing behavior
2) have an option NOT to disable the stream, e.g. `disable_stream` - defaults to `true`, disabling the stream to preserve existing behavior

Because in many use cases that would follow what I'm doing, disabling the stream is completely unnecessary, as the PIN is already set. Then I'd need to re-enable the stream, which is again an unnecessary request.

Reply:
> Because in many use cases that would follow what I'm doing, disabling the stream is completely unnecessary, as the PIN is already set. Then I'd need to re-enable the stream, which is again an unnecessary request.

Ah yes, I hadn't considered this. In that case, I guess a request that just kicks viewers makes sense. You still need to clarofy in both the documentation and code that this won't work for all types of mountpoints, though: on demand mountpoints for instance don't have a group of viewers, but spawn a different thread for each of them, since each has a separate context of the same source media.

Reply:
Or altervatively `disable_stream` could be called `close_only`, defaulting to `false` which would indicate that connections should be closed only.

What I'm after is forcing users to perform unnecessary operations here. Doesn't really matter what they're called :)

Reply:
No, that wouldn't be a `disable` then, it would be highjacking a different request to then only do something else. I guess kick is fine :smile:

Reply:
Ok, so I'll be keeping the separate `kick_all` request after all now :)

Can you expand a bit more on what to put into the documentation? I have to say I'm not aware how the plugin works in these different cases. Or alternatively, could we make `kick_all` work for on-demand mountpoints too?

Reply:
I'd simply say that the request works with all mountpoint types except the ondemand ones. I wouldn't waste time trying to get the method to work with those too, since they don't use the viewers list as the others do, and don't keep track of who's receiving it: each viewer has their own thread and their own playback context.

Reply:
:+1: Done!